source/actions/SaveAllSteppingAction.cc

```diff
@@ -63,6 +63,8 @@ void SaveAllSteppingAction::UserSteppingAction(const G4Step* step)
 
   G4ThreeVector initial_pos = pre ->GetPosition();
   G4ThreeVector   final_pos = post->GetPosition();
+  G4double        step_time = (pre->GetGlobalTime()  +
```

Comment:
Are you sure that you want the average time of the steps and not the time at the beginning or at the end? What do you use this for?

Reply:
I don't have a preference. I just want to have a handle on the time of the steps so I can potentially build a waveform out of them. Saving both pre and post seemed a bit too much, but I don't mind doing that and taking the average offline.

Reply:
It's just that in G4 the "borders" of a step are well defined in terms of position, volume... while what happens "inside" the step is sometimes more confuse. However, since yours is the only user case so far and you want the average, we can go ahead with that.